tutor/hooks/consts.py

```diff
@@ -171,21 +187,27 @@ def your_filter(items):
     #:      rendered at runtime with the user configuration. Thus, the image tag could
     #:      be ``"{{ DOCKER_REGISTRY }}/myimage:{{ TUTOR_VERSION }}"``.
     #:    - ``args`` is a list of arguments that will be passed to ``docker build ...``.
-    #: :parameter dict config: user configuration.
-    IMAGES_BUILD = filters.get("images:build")
+    #: :parameter Config config: user configuration.
+    IMAGES_BUILD: "Filter[t.List[t.Tuple[str, t.Tuple[str, ...], str, t.Tuple[str, ...]]], [Config]]" = filters.get(
```

Comment:
We could put `from __future__ import annotations` at the top of this module, which enables deferred annotation evaluation for Python 3.7+ and would allow us to drop these quotation marks.
```suggestion
    IMAGES_BUILD: Filter[t.List[t.Tuple[str, t.Tuple[str, ...], str, t.Tuple[str, ...]]], [Config]] = filters.get(
```
Even better, given that we're requiring Python 3.9+ now for type checking, we could start subscripting types directly instead of using the `typing` module:
```suggestion
    IMAGES_BUILD: Filter[list[tuple[str, tuple[str, ...], str, tuple[str, ...]]], [Config]]" = filters.get(
```
I'm also happy make that second refactoring in a separate PR, since it'd be a pretty big diff.

Reply:
OK this is very strange. `from __future__ import annotations` is already present at the top of this file. I can't remember at all why I added it in the first place. You are absolutely right that this future module allows us to remove the quotation marks. I had not realised that.

So, great news! We can simplify the type definitions.